spec/requests/games_spec.rb

```diff
@@ -30,56 +30,64 @@
+                                                       'symbol' => ["can't be blank"] })
+      end
+
+      it 'validates board size to be integer' do
+        game_params[:board_size] = 'X'
+        post '/games', params: game_params, as: :json
+        expect(response.parsed_body['errors']).to eq({ 'board' => ["can't be blank"] })
+      end
+
+      it 'validates the game_mode to either be 1 or 2' do
+        game_params[:game_mode] = 3
+        post '/games', params: game_params, as: :json
+        expect(response.parsed_body['errors']).to eq('game_mode' => ['is not included in the list'])
+        game_params[:game_mode] = 2
+        post '/games', params: game_params, as: :json
+        expect(response).to have_http_status(:created)
+      end
+    end
   end
 
   describe '#play' do
     let(:find_game) { instance_double(FindGameService) }
-    let(:game_engine) { instance_double(CreateWebGameEngineService) }
-    let(:play_params) do
-      {
-        "move": '1'
-      }
-    end
+    let(:create_game_engine) { instance_double(CreateWebGameEngineService) }
+    let(:web_engine) { instance_double(TicTacToe::WebEngine) }
     let(:game_params) do
       {
         "language": 'fr',
         "player_name": 'Manzi',
-        "game_mode": '2',
-        "board_size": '3',
+        "game_mode": 2,
+        "board_size": 3,
         "symbol": 'X'
       }
     end
     let(:game) { create(:game) }
-    it 'returns play ongoing if not finished' do
-      put "/games/#{game.id}/play", params: play_params, as: :json
+    it 'returns game state after applying move' do
+      put "/games/#{game.id}/play", params: { "move": 1 }, as: :json
       expect(response.parsed_body['state']).to eq('Ongoing')
     end
 
-    context 'tests if there is a draw' do
-      it 'returns draw message if the game is tied' do
-        game.board = [['-', 'O', 'X'],
-                      %w[X O O],
-                      %w[O X X]]
-        game.save
-        put "/games/#{game.id}/play", params: { "move": '1' }, as: :json
-        expect(response.parsed_body['state']).to eq("It's a Draw")
+    describe 'Validations' do
+      it 'validates the game id' do
+        id = 0
+        put "/games/#{id}/play", params: { "move": 1 }, as: :json
+        expect(response.parsed_body['errors']).to eq("Couldn't find Game with 'id'=#{id}")
       end
-    end
-    context 'tests if there is a winner' do
-      let(:moves) do
-        [{
-          "move": '1'
-        }, {
-          "move": '2'
-        }, {
-          "move": '4'
-        }]
-      end
-      it 'returns winner message if the game is won' do
-        moves.each do |play_params|
-          put "/games/#{game.id}/play", params: play_params, as: :json
-        end
-        expect(response.parsed_body['state']).to eq('Intelligent Computer won the game')
+
+      it 'validates the player move' do
+        put "/games/#{game.id}/play", params: { "move": 10 }, as: :json
+        p response.parsed_body
```

Comment:
Remove the debug statement.

Reply:
ec09f85876e38985cf84fe1735fc1c224a44443e